certs/certs.go

```diff
@@ -99,10 +109,33 @@ func ValidateRoot(certStore trustmanager.X509Store, root *data.Signed, gun strin
 			return &ErrValidationFail{Reason: "failed to validate data with current trusted certificates"}
 		}
 	} else {
-		logrus.Debugf("found no currently valid root certificates for %s", gun)
+		logrus.Debugf("found no currently valid root certificates for %s, using trust_pinning config to bootstrap trust:", gun, trustPinning)
+		trustPinChecker, err := NewTrustPinChecker(trustPinning, gun)
+		if err != nil {
+			return &ErrValidationFail{Reason: err.Error()}
+		}
+
+		validPinnedCerts := []*x509.Certificate{}
+		for _, cert := range allValidCerts {
+			_, intermediateCerts := parseAllCerts(signedRoot)
```

Comment:
`intermediateCerts` seems like it remains the same each time, since `parseAllCerts` takes just the `signedRoot`.  Can this just be called once, rather than for every cert in `allValidCerts`?  Unless verifying a cert given the roots and intermediates mutates those cert pools.

Also I think `allValidCerts` calls `parseAllCerts` as well, and ignores the intermediate certs - since we just seem to call it here, I'm wondering if we can refactor `validatRootLeafCerts` to take a bunch of leaf certs rather than the `data.SignedRoot`?  That way all that stuff can just be calculated in this if/else block?